### PR TITLE
[WIP] Validate PR with GitHub Actions.

### LINKED ISSUE #58: Warn on CI if esy.lock is not up to date
If any of the esy.lock's is not up to date we should warn.

Additionally I'm wondering if it's possible to put all the locks and package-ci-*.jsons in a `ci` directory and have it still work.

### PR DESCRIPTION
**Pull request by Et7f3**
This PR re-open #52 (because it started with a fork of @ulrikstrid branch)

> Most notably Github Actions' cache is too limited. 500 megabytes is not enough for esy!
> 
>_Originally posted by @wokalski in https://github.com/briskml/brisk-reconciler/pull/52#issuecomment-554788105_

With #59 the cache tarball go from 420 Mo to 360Mo and can be cached. (see my actions run it run in 2 minutes after downloading esy)

https://github.com/briskml/brisk-reconciler/commit/d185e75c47b381fb911ca9decdf7525b97188ea1 fix #58 

What should be done
- [ ] Merge [npm-publish.yml](https://github.com/briskml/brisk-reconciler/blob/0d6836bb67bd74cee16ff401de6f83147eb20f05/.github/workflows/npm-publish.yml)
- [ ] Add stale issue detection (just a message ?)
- [x] Check that formatting is correct.
- [x] Hook up [release-drafter](https://github.com/release-drafter/release-drafter)

### CODE REVIEW

**Comment by Et7f3** [comment]
Ah release drafter need `.github/release-drafter.yml` on master. You can see in my fork where I have merged the result.

**Comment by Et7f3** [comment]
:smile: You can see the last commit was triggered by Mac OS runner. When it push change it make other runner to fail, but GitHub relaunch the runner because a new commit is done. I need a way to make it work also for windows :cry: but it might be cool. We will never forget to commit lock files. So it doesn't warn like #58 suggested it also update it :smiley:.

**Comment by Et7f3** [comment]
The same can be done for formatting but it may render diff larger. What do you think ?

File: .github/workflows/validate-pr.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ jobs:
       - name: Install dependencies
         run: esy install -P package-ci-${{ matrix.ocaml_version }}
       - name: Check that lock directory are up-to-date. If this fails, run `esy install -P pacage-ci-${{ matrix.ocaml_version }}` and commit changes and push again.
-        run: git diff --exit-code
+        run: git diff --exit-code || (git add package-ci-${{ matrix.ocaml_version }}.esy.lock "*.lock"; git push origin $GITHUB_REF)
```
Comment (wokalski) [inline .github/workflows/validate-pr.yml]: This is very nice 👍

**Comment by wokalski** [comment]
Thanks for the PR @Et7f3. I really dig having CI on github actions but tbh I don't see a good reason to switch from azure which is more robust. When GA pulls out the cache limit we might revisit this decision. We are not using any of the advanced Azure features and I don't think we will.

Building aside, hooking up things like `release-drafter` together with a nicer release workflow (preferably where npm publish goes together with publishing the github release) does make sense to me. I don't think there are good alternatives for azure and frankly, the output is awesome.

What do you think?

**Comment by wokalski** [comment]
Regarding formatting, I think it's a good idea to add auto format. Pushing commits from CI is sort of controversial but overall it'll be a quality of life improvement 👍.

Can we add a message from a bot like:

```
I've updated your test lock files and formatted the code. Please pull the last commit before pushing any more changes.
```

**Comment by wokalski** [comment]
And lastly, we could similar verification step for opam file as discussed in #66.

**Comment by Et7f3** [comment]
2 dependencies less.

**Comment by Et7f3** [comment]
> And lastly, we could similar verification step for opam file as discussed in #66.

Fixed in https://github.com/briskml/brisk-reconciler/pull/69/commits/508851d5f9ce283d9d9b2dc1d054862921829c35

**Comment by Et7f3** [comment]
In last I use the advanced template so `feature`, `enhancement`, `fix`, `bugfix`, `bug`, `chore` label can be added.

**Comment by Et7f3** [comment]
Ok I get 403 because it is not merged but it work fine in my fork https://github.com/Et7f3/brisk-reconciler/pull/2#issuecomment-562257624

**Comment by Et7f3** [comment]
> Regarding formatting, I think it's a good idea to add auto format. Pushing commits from CI is sort of controversial but overall it'll be a quality of life improvement +1.
> 
> Can we add a message from a bot like:
> 
> ```
> I've updated your test lock files and formatted the code. Please pull the last commit before pushing any more changes.
> ```

Done :)

**Comment by Et7f3** [comment]
You can customize message if you want :) 
image

**Comment by wokalski** [comment]
That's extremely good. One thing, could you please organise the commits a bit more? Or should I just squash it all? There are quite a few changes. I suppose most of them are lock file related but nevertheless 🙂.

**Comment by Et7f3** [comment]
Lot of commits come from merge I will try rebasing so this noise disapear.

**Pull request closed without merging**